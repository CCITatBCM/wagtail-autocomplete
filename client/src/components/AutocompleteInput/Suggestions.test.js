import React from 'react';
import { shallow } from 'enzyme';

import Suggestions from './Suggestions';


const mockProps = {
  suggestions: [
    { id: 1, title: 'Alice' },
    { id: 2, title: 'Tekisha' },
  ],
  onChange: jest.fn(),
  onCreate: jest.fn(),
  onClick: jest.fn(),
  input: '',
  canCreate: true,
};


describe('Suggestions', () => {
  it('exists', () => {
    expect(Suggestions).toBeDefined();
  });

  it('mounts', () => {
    const suggestions = shallow(
      <Suggestions {...mockProps} />
    );
    expect(suggestions).toMatchSnapshot();
  });
});
