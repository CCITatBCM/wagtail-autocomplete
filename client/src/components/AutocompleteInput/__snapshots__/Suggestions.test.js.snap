// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestions mounts 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <span
    className="c-wagtailautocomplete__suggestions__sub"
>
    <div
        className="c-wagtailautocomplete__search-input-container"
    >
        <input
            className="c-wagtailautocomplete__search"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
        />
        <svg
            className="c-wagtailautocomplete__search-icon"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
        >
            <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                fill="currentColor"
            />
            <path
                d="M0 0h24v24H0z"
                fill="none"
            />
        </svg>
    </div>
    <ul
        className="c-wagtailautocomplete__suggestions"
        style={
            Object {
                "display": "none",
              }
        }
    >
        <li
            className="c-wagtailautocomplete__suggestions__item c-wagtailautocomplete__suggestions__item--active"
            onClick={[Function]}
            onMouseEnter={[Function]}
        >
            <span>
                Alice
            </span>
            <svg
                viewBox="0 0 1792 1792"
                xmlns="http://www.w3.org/2000/svg"
            >
                <path
                    d="M1600 960q0 54-37 91l-651 651q-39 37-91 37-51 0-90-37l-75-75q-38-38-38-91t38-91l293-293h-704q-52 0-84.5-37.5t-32.5-90.5v-128q0-53 32.5-90.5t84.5-37.5h704l-293-294q-38-36-38-90t38-90l75-75q38-38 90-38 53 0 91 38l651 651q37 35 37 90z"
                />
            </svg>
        </li>
        <li
            className="c-wagtailautocomplete__suggestions__item"
            onClick={[Function]}
            onMouseEnter={[Function]}
        >
            <span>
                Tekisha
            </span>
            <svg
                viewBox="0 0 1792 1792"
                xmlns="http://www.w3.org/2000/svg"
            >
                <path
                    d="M1600 960q0 54-37 91l-651 651q-39 37-91 37-51 0-90-37l-75-75q-38-38-38-91t38-91l293-293h-704q-52 0-84.5-37.5t-32.5-90.5v-128q0-53 32.5-90.5t84.5-37.5h704l-293-294q-38-36-38-90t38-90l75-75q38-38 90-38 53 0 91 38l651 651q37 35 37 90z"
                />
            </svg>
        </li>
        <li
            className="c-wagtailautocomplete__suggestions__item c-wagtailautocomplete__suggestions__item--create"
            onClick={[Function]}
            onMouseEnter={[Function]}
        >
            Create new “
            ”
        </li>
    </ul>
</span>,
  "nodes": Array [
    <span
      className="c-wagtailautocomplete__suggestions__sub"
>
      <div
            className="c-wagtailautocomplete__search-input-container"
      >
            <input
                  className="c-wagtailautocomplete__search"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  type="text"
            />
            <svg
                  className="c-wagtailautocomplete__search-icon"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
            >
                  <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                        fill="currentColor"
                  />
                  <path
                        d="M0 0h24v24H0z"
                        fill="none"
                  />
            </svg>
      </div>
      <ul
            className="c-wagtailautocomplete__suggestions"
            style={
                  Object {
                        "display": "none",
                      }
            }
      >
            <li
                  className="c-wagtailautocomplete__suggestions__item c-wagtailautocomplete__suggestions__item--active"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
            >
                  <span>
                        Alice
                  </span>
                  <svg
                        viewBox="0 0 1792 1792"
                        xmlns="http://www.w3.org/2000/svg"
                  >
                        <path
                              d="M1600 960q0 54-37 91l-651 651q-39 37-91 37-51 0-90-37l-75-75q-38-38-38-91t38-91l293-293h-704q-52 0-84.5-37.5t-32.5-90.5v-128q0-53 32.5-90.5t84.5-37.5h704l-293-294q-38-36-38-90t38-90l75-75q38-38 90-38 53 0 91 38l651 651q37 35 37 90z"
                        />
                  </svg>
            </li>
            <li
                  className="c-wagtailautocomplete__suggestions__item"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
            >
                  <span>
                        Tekisha
                  </span>
                  <svg
                        viewBox="0 0 1792 1792"
                        xmlns="http://www.w3.org/2000/svg"
                  >
                        <path
                              d="M1600 960q0 54-37 91l-651 651q-39 37-91 37-51 0-90-37l-75-75q-38-38-38-91t38-91l293-293h-704q-52 0-84.5-37.5t-32.5-90.5v-128q0-53 32.5-90.5t84.5-37.5h704l-293-294q-38-36-38-90t38-90l75-75q38-38 90-38 53 0 91 38l651 651q37 35 37 90z"
                        />
                  </svg>
            </li>
            <li
                  className="c-wagtailautocomplete__suggestions__item c-wagtailautocomplete__suggestions__item--create"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
            >
                  Create new “
                  ”
            </li>
      </ul>
</span>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Suggestions
        canCreate={true}
        input=""
        onChange={[Function]}
        onClick={[Function]}
        onCreate={[Function]}
        suggestions={
                Array [
                        Object {
                          "id": 1,
                          "title": "Alice",
                        },
                        Object {
                          "id": 2,
                          "title": "Tekisha",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Suggestions {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleBlurClick": [Function],
        "handleFocus": [Function],
        "handleKeyPress": [Function],
        "props": Object {
          "canCreate": true,
          "input": "",
          "onChange": [Function],
          "onClick": [Function],
          "onCreate": [Function],
          "suggestions": Array [
            Object {
              "id": 1,
              "title": "Alice",
            },
            Object {
              "id": 2,
              "title": "Tekisha",
            },
          ],
        },
        "refs": Object {},
        "state": Object {
          "index": 0,
          "visible": false,
        },
        "suggestionItemsElm": Object {},
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <span
          className="c-wagtailautocomplete__suggestions__sub"
>
          <div
                    className="c-wagtailautocomplete__search-input-container"
          >
                    <input
                              className="c-wagtailautocomplete__search"
                              onChange={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              type="text"
                    />
                    <svg
                              className="c-wagtailautocomplete__search-icon"
                              viewBox="0 0 24 24"
                              xmlns="http://www.w3.org/2000/svg"
                    >
                              <path
                                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                                        fill="currentColor"
                              />
                              <path
                                        d="M0 0h24v24H0z"
                                        fill="none"
                              />
                    </svg>
          </div>
          <ul
                    className="c-wagtailautocomplete__suggestions"
                    style={
                              Object {
                                        "display": "none",
                                      }
                    }
          >
                    <li
                              className="c-wagtailautocomplete__suggestions__item c-wagtailautocomplete__suggestions__item--active"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                    >
                              <span>
                                        Alice
                              </span>
                              <svg
                                        viewBox="0 0 1792 1792"
                                        xmlns="http://www.w3.org/2000/svg"
                              >
                                        <path
                                                  d="M1600 960q0 54-37 91l-651 651q-39 37-91 37-51 0-90-37l-75-75q-38-38-38-91t38-91l293-293h-704q-52 0-84.5-37.5t-32.5-90.5v-128q0-53 32.5-90.5t84.5-37.5h704l-293-294q-38-36-38-90t38-90l75-75q38-38 90-38 53 0 91 38l651 651q37 35 37 90z"
                                        />
                              </svg>
                    </li>
                    <li
                              className="c-wagtailautocomplete__suggestions__item"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                    >
                              <span>
                                        Tekisha
                              </span>
                              <svg
                                        viewBox="0 0 1792 1792"
                                        xmlns="http://www.w3.org/2000/svg"
                              >
                                        <path
                                                  d="M1600 960q0 54-37 91l-651 651q-39 37-91 37-51 0-90-37l-75-75q-38-38-38-91t38-91l293-293h-704q-52 0-84.5-37.5t-32.5-90.5v-128q0-53 32.5-90.5t84.5-37.5h704l-293-294q-38-36-38-90t38-90l75-75q38-38 90-38 53 0 91 38l651 651q37 35 37 90z"
                                        />
                              </svg>
                    </li>
                    <li
                              className="c-wagtailautocomplete__suggestions__item c-wagtailautocomplete__suggestions__item--create"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                    >
                              Create new “
                              ”
                    </li>
          </ul>
</span>,
        "_debugID": 2,
        "_renderedOutput": <span
          className="c-wagtailautocomplete__suggestions__sub"
>
          <div
                    className="c-wagtailautocomplete__search-input-container"
          >
                    <input
                              className="c-wagtailautocomplete__search"
                              onChange={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              type="text"
                    />
                    <svg
                              className="c-wagtailautocomplete__search-icon"
                              viewBox="0 0 24 24"
                              xmlns="http://www.w3.org/2000/svg"
                    >
                              <path
                                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                                        fill="currentColor"
                              />
                              <path
                                        d="M0 0h24v24H0z"
                                        fill="none"
                              />
                    </svg>
          </div>
          <ul
                    className="c-wagtailautocomplete__suggestions"
                    style={
                              Object {
                                        "display": "none",
                                      }
                    }
          >
                    <li
                              className="c-wagtailautocomplete__suggestions__item c-wagtailautocomplete__suggestions__item--active"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                    >
                              <span>
                                        Alice
                              </span>
                              <svg
                                        viewBox="0 0 1792 1792"
                                        xmlns="http://www.w3.org/2000/svg"
                              >
                                        <path
                                                  d="M1600 960q0 54-37 91l-651 651q-39 37-91 37-51 0-90-37l-75-75q-38-38-38-91t38-91l293-293h-704q-52 0-84.5-37.5t-32.5-90.5v-128q0-53 32.5-90.5t84.5-37.5h704l-293-294q-38-36-38-90t38-90l75-75q38-38 90-38 53 0 91 38l651 651q37 35 37 90z"
                                        />
                              </svg>
                    </li>
                    <li
                              className="c-wagtailautocomplete__suggestions__item"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                    >
                              <span>
                                        Tekisha
                              </span>
                              <svg
                                        viewBox="0 0 1792 1792"
                                        xmlns="http://www.w3.org/2000/svg"
                              >
                                        <path
                                                  d="M1600 960q0 54-37 91l-651 651q-39 37-91 37-51 0-90-37l-75-75q-38-38-38-91t38-91l293-293h-704q-52 0-84.5-37.5t-32.5-90.5v-128q0-53 32.5-90.5t84.5-37.5h704l-293-294q-38-36-38-90t38-90l75-75q38-38 90-38 53 0 91 38l651 651q37 35 37 90z"
                                        />
                              </svg>
                    </li>
                    <li
                              className="c-wagtailautocomplete__suggestions__item c-wagtailautocomplete__suggestions__item--create"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                    >
                              Create new “
                              ”
                    </li>
          </ul>
</span>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Suggestions
    canCreate={true}
    input=""
    onChange={[Function]}
    onClick={[Function]}
    onCreate={[Function]}
    suggestions={
        Array [
            Object {
              "id": 1,
              "title": "Alice",
            },
            Object {
              "id": 2,
              "title": "Tekisha",
            },
          ]
    }
/>,
}
`;
